spec.md

```diff
@@ -639,28 +731,36 @@ float(3) / 2                                    # 1.5
 
 ### Strings
 
-A string represents an immutable sequence of bytes.
+A string is an immutable sequence of elements that encode Unicode text.
 The [type](#type) of a string is `"string"`.
 
-Strings can represent arbitrary binary data, including zero bytes, but
-most strings contain text, encoded by convention using UTF-8.
+Depending on the implementation, the elements may be 8- or 16-bit values,
```

Comment:
I'm not sure what the intended use of these strings is, if they may contain invalid data, and they index the same as `bytes`... What's the goal of having two distinct types here, where their guarantees and API surface are ~identical? Is it just to give the Java implementation scope to use UTF-16 to represent strings?

Reply:
Concretely, I guess my question really is: Should strings perhaps guarantee they contain valid UTF-K data, and perhaps index or at least allow views to iterate over codepoints rather than k-byte elements? (Or is some of this the "still to come" part? :))